systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -0,0 +1,162 @@
+        LOGGER.info("Modifying KafkaNodePool: {}/{} by increasing number of Kafka replicas from '3' to '4'", testStorage.getNamespaceName(), testStorage.getKafkaNodePoolName());
+        KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(testStorage.getKafkaNodePoolName(), kafkaNodePool -> {
+            kafkaNodePool.getSpec().setReplicas(4);
+        }, testStorage.getNamespaceName());
+
+        waitForPodSetKafkaPodsReady(testStorage.getNamespaceName(), 4, testStorage.getKafkaStatefulSetName());
+
+        LOGGER.info("Producing and Consuming messages with clients: {}, {} in Namespace {}", testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName());
+        resourceManager.createResource(extensionContext,
+            clients.producerStrimzi(),
+            clients.consumerStrimzi()
+        );
+        ClientUtils.waitForClientsSuccess(testStorage);
+
+        LOGGER.info("Transferring Kafka Cluster: {}/{} from KafkaNodePool enabled, to being disabled by removing annotation", testStorage.getNamespaceName(), testStorage.getClusterName());
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> {
+                kafka.getMetadata().getAnnotations().clear();
+            }, testStorage.getNamespaceName());
+
+        waitForPodSetKafkaPodsReady(testStorage.getNamespaceName(), 3, KafkaResources.kafkaStatefulSetName(testStorage.getClusterName()));
+
+        LOGGER.info("Producing and Consuming messages with clients: {}, {} in Namespace {}", testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName());
+        resourceManager.createResource(extensionContext,
+            clients.producerStrimzi(),
+            clients.consumerStrimzi()
+        );
+        ClientUtils.waitForClientsSuccess(testStorage);
+
+        LOGGER.info("Transferring Kafka Cluster: {}/{} from KafkaNodePool enabled, to being disabled by removing annotation", testStorage.getNamespaceName(), testStorage.getClusterName());
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> {
+                kafka.getMetadata().getAnnotations().put(Annotations.ANNO_STRIMZI_IO_NODE_POOLS, "enabled");
+            }, testStorage.getNamespaceName());
+
+        waitForPodSetKafkaPodsReady(testStorage.getNamespaceName(), 4, testStorage.getKafkaStatefulSetName());
+
+        LOGGER.info("Producing and Consuming messages with clients: {}, {} in Namespace {}", testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName());
+        resourceManager.createResource(extensionContext,
+            clients.producerStrimzi(),
+            clients.consumerStrimzi()
+        );
+        ClientUtils.waitForClientsSuccess(testStorage);
+
+    }
+
+    private static void waitForPodSetKafkaPodsReady(String namespace, int expectedPodsCount, String controllingPodSetName) {
```

Comment:
This logic should be wrapped inside *utils class (e.g., PodUtils).

Reply:
i decided to remove that method alltogether but good catch thanks .